src/StelLogger.cpp

```diff
@@ -44,193 +78,352 @@ void StelLogger::init(const QString& logFilePath)
+
+	int ncpu = 0;
+	len = sizeof(ncpu);
+	sysctlbyname("hw.ncpu", &ncpu, &len, nullptr, 0);
+	writeLog(QString("Processor logical cores: %1").arg(ncpu));
+
+	// memory info
+	int64_t totalRAM = 0;
+	len = sizeof(totalRAM);
+	sysctlbyname(_physmem, &totalRAM, &len, nullptr, 0);
+	writeLog(QString("Total physical memory: %1 MB").arg(totalRAM/(1024<<10)));
+#endif
+
+#ifdef Q_OS_OPENBSD
+	int mib[2], freq, ncpu;
+	size_t len = 1024;
+	std::string model;
+	model.resize(len);
+
+	// CPU info
+	mib[0] = CTL_HW;
+	mib[1] = HW_MODEL;
+	sysctl(mib, 2, model.data(), &len, NULL, 0);
+	model.resize(len);
+	writeLog(QString("Processor name: %1").arg(model.data()));
+
+	mib[0] = CTL_HW;
+	mib[1] = HW_CPUSPEED;
+	len = sizeof(freq);
+	sysctl(mib, 2, &freq, &len, NULL, 0);
+	writeLog(QString("Processor speed: %1 MHz").arg(freq));
+
+	mib[0] = CTL_HW;
+	mib[1] = HW_NCPU;
+	len = sizeof(ncpu);
+	sysctl(mib, 2, &ncpu, &len, NULL, 0);
+	writeLog(QString("Processor logical cores: %1").arg(ncpu));
+
+	// memory info
+	mib[0] = CTL_HW;
+	#if defined(HW_PHYSMEM64)
+	mib[1] = HW_PHYSMEM64;
 	#else
-	QStringList systemLines = systemData.split('\n', QString::SkipEmptyParts);
+	mib[1] = HW_PHYSMEM;
 	#endif
-	for (int i = 0; i<systemLines.size(); i++)
-	{
-		// hardware overview
-		if(systemLines.at(i).contains("Model", Qt::CaseInsensitive))
-			writeLog(systemLines.at(i).trimmed());
+	int64_t totalRAM = 0;
+	len = sizeof(totalRAM);
+	sysctl(mib, 2, &totalRAM, &len, NULL, 0);
+	writeLog(QString("Total physical memory: %1 MB").arg(totalRAM/(1024<<10)));
+#endif
 
-		if(systemLines.at(i).contains("Chip:", Qt::CaseInsensitive))
-			writeLog(systemLines.at(i).trimmed());
+#ifdef Q_OS_SOLARIS
+	processor_info_t pinfo;
+	processor_info(0, &pinfo);
+	//writeLog(QString("Processor name: %1").arg(pinfo.pi_processor_type));
+	writeLog(QString("Processor speed: %1 MHz").arg(pinfo.pi_clock));
 
-		if(systemLines.at(i).contains("Processor", Qt::CaseInsensitive))
-			writeLog(systemLines.at(i).trimmed().replace("Unknown", "Apple M1"));
+	int ncpu = sysconf( _SC_NPROCESSORS_ONLN );
+	writeLog(QString("Processor logical cores: %1").arg(ncpu));
 
-		if(systemLines.at(i).contains("Memory", Qt::CaseInsensitive))
-			writeLog(systemLines.at(i).trimmed());
+	// memory info
+	int64_t totalRAM = (size_t)sysconf( _SC_PHYS_PAGES ) * (size_t)sysconf( _SC_PAGESIZE );
+	writeLog(QString("Total physical memory: %1 MB").arg(totalRAM/(1024<<10)));
+#endif
 
-		// graphics/display overview
-		if(systemLines.at(i).contains("Resolution", Qt::CaseInsensitive))
-			writeLog(systemLines.at(i).trimmed());
+#ifdef Q_OS_HAIKU
+	// Idea and some code is catching from Haiku OS (MIT license)
+	uint32 topologyNodeCount = 0;
+	cpu_topology_node_info* topology = NULL;
 
-		if(systemLines.at(i).contains("VRAM", Qt::CaseInsensitive))
-			writeLog(systemLines.at(i).trimmed());
-	}
+	get_cpu_topology_info(NULL, &topologyNodeCount);
+	if (topologyNodeCount != 0)
+		topology = new cpu_topology_node_info[topologyNodeCount];
```

Comment:
Better use a smart pointer here, e.g. `std::unique_ptr`.

Reply:
Why? I don't think using std::unique_ptr is better here

Reply:
Because generally it's better to avoid the need to worry about `delete` or `delete[]`, and let the compiler (destructor) do this for you. In fact, I didn't think trough the case here, the best would be to use `std::vector<cpu_topology_node_info>`, since this would show the semantics better.

Reply:
`get_cpu_topology_info` method can work with standard pointers, but not `std::unique_ptr` nor `std::shared_ptr` - I don't want to touch a haiku kernel.

Reply:
You don't need to touch the dependencies. Both `unique_ptr` and `shared_ptr` (which would be a worse choice) provide a `get()` method that returns a raw pointer that you can pass to any C API without giving up ownership. Same for Qt's smart pointers (since Qt 5.11).
If you use `std::vector` instead, then the method to get a pointer is `std::vector::data()` or simply `&myvector[0]`.